crates/core/tedge_actors/src/builders.rs

```diff
@@ -213,82 +197,6 @@ pub trait RuntimeRequestSink {
     fn get_signal_sender(&self) -> DynSender<RuntimeRequest>;
 }
 
-/// A trait that defines that an actor provides a service
-/// by accepting `Request` messages and sending `Response` from/to its peers.
-///
-/// In order to connect to a to `ServiceProvider<Req, Res, Conf>` and avail its services,
-/// the peer must be a `ServiceConsumer<Req, Res, Conf>`.
-///
-/// The config parameter is typically used by the `ServiceConsumer`
-/// to register any message filtering criteria to the `ServiceProvider`.
-pub trait ServiceProvider<Request: Message, Response: Message, Config> {
-    /// Connect a peer message box to the message box under construction
-    fn add_peer(&mut self, peer: &mut impl ServiceConsumer<Request, Response, Config>) {
-        let config = peer.get_config();
-        let response_sender = peer.get_response_sender();
-        let request_sender = self.connect_consumer(config, response_sender);
-        peer.set_request_sender(request_sender);
-    }
-
-    /// Connect a consumer to the service provider under construction
-    /// returning to that service consumer a sender for its requests.
-    ///
-    /// The consumer provides:
-    /// - a config to filter the responses of interest,
-    /// - a sender where the responses will have to be sent by the service.
-    ///
-    /// The consumer is given back:
-    /// - a sender where its requests will have to be sent to the service.
-    fn connect_consumer(
-        &mut self,
-        config: Config,
-        response_sender: DynSender<Response>,
-    ) -> DynSender<Request>;
-}
-
-/// A trait that defines that the actor under-construction
-/// is a consumer of the service provided by another actor that is a `ServiceProvider`.
-///
-/// A `ServiceConsumer<Req, Res, Conf>` actor can be connected to another actor as its peer
-/// if that actor is a `ServiceProvider<Req, Res, Conf>`.
-pub trait ServiceConsumer<Request: Message, Response: Message, Config> {
-    /// Return the config used by this actor to connect the service provider
-    fn get_config(&self) -> Config;
-
-    /// Set the sender to be used by this actor's box to send requests
-    fn set_request_sender(&mut self, request_sender: DynSender<Request>);
-
-    /// Return a sender where the responses to this actor's box have to be sent
-    fn get_response_sender(&self) -> DynSender<Response>;
-
-    /// Connect this client message box to the service message box
-    fn set_connection(
-        &mut self,
-        service: &mut impl ServiceProvider<Request, Response, Config>,
-    ) -> &mut Self
-    where
-        Self: Sized,
-    {
-        service.add_peer(self);
-        self
-    }
```

Comment:
As the ServiceProvider and ServiceConsumer traits are deprecated, this convenient method can no more be used.

One has now to be explicit.

A connection to MQTT is bidirectional:

```
        let mqtt_publisher = mqtt.get_sender(); 
        mqtt.register_peer(mqtt_subscriptions, message_box.get_sender());
```

To connect a server, one can either use a `ClienbtMessageBox` to await the response of each request:

```
        let download_sender = ClientMessageBox::new(downloader_actor);
```

or to redirect the responses to another message box:

```
        let timer_sender = timer.add_requester(box_builder.get_sender());
```